middleware/u8glib/csrc/u8g_com_embarc_ssd1306_spi.c

```diff
@@ -0,0 +1,275 @@
+/*
+
+  u8g.h
+
+  Universal 8bit Graphics Library
+
+  Copyright (c) 2011, olikraus@gmail.com
+  All rights reserved.
+
+  Redistribution and use in source and binary forms, with or without modification,
+  are permitted provided that the following conditions are met:
+
+  * Redistributions of source code must retain the above copyright notice, this list
+    of conditions and the following disclaimer.
+
+  * Redistributions in binary form must reproduce the above copyright notice, this
+    list of conditions and the following disclaimer in the documentation and/or other
+    materials provided with the distribution.
+
+  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND
+  CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+  INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+  MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
+  NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+  STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+*/
+/*
+ * Modified for port to ARC embARC platform
+ * by Wayne Ren wei.ren@synopsys.com, Synopsys, Inc.
+ */
+
+#include "u8g.h"
+#include "embARC.h"
+#include "dev_gpio.h"
+#include "ssd1306_app_config.h"
+
+#ifndef SSD1306_SPI_ID
+#define SSD1306_SPI_ID		0
+#endif
+
+#ifndef SSD1306_SPI_CS
+#define SSD1306_SPI_CS		DEV_SPI_CS_LINE_1
+#endif
+
+#ifndef SSD1306_USE_CS_PIN
+	#define SSD1306_USE_CS_PIN	1
+#endif
+
+#if SSD1306_USE_CS_PIN
+# ifndef SSD1306_CS_GPIO
+	#define SSD1306_CS_GPIO	(DEV_GPIO_PORT_A)
+# endif
+
+# ifndef SSD1306_CS_PIN
+	#define SSD1306_CS_PIN		24
+# endif
+static void oled_ssd1306_chip_select(DEV_SPI* _oled_spi, bool select)
+{
+	static DEV_GPIO *oled_gpio_cs;
```

Comment:
You need to change static DEV_GPIO *oled_gpio_cs; to static DEV_GPIO *oled_gpio_cs = NULL; since the initial value may not be NULL